rust/main/agents/validator/src/submit.rs

```diff
@@ -343,13 +345,27 @@ impl ValidatorSubmitter {
             "Fetched checkpoint from checkpoint storage",
         );
 
-        if existing.is_some() {
-            debug!(index = checkpoint.index, "Checkpoint already submitted");
-            return Ok(());
+        if let Some(existing) = existing.as_ref() {
+            let existing_signer = existing.recover()?;
+            let signer = self.signer.eth_address();
+            if existing_signer == signer && existing.value == checkpoint {
+                debug!(index = checkpoint.index, "Checkpoint already submitted");
+                return Ok(());
+            } else {
+                warn!(
+                    index = checkpoint.index,
+                    existing_checkpoint = ?existing.value,
+                    existing_signer = ?existing_signer,
+                    new_checkpoint = ?checkpoint,
+                    new_signer = ?signer,
+                    "Checkpoint already submitted, but with different values, overwriting"
+                );
+            }
         }
 
-        let start = Instant::now();
         let signed_checkpoint = self.sign_checkpoint(checkpoint).await?;
+
+        let start = Instant::now();
```

Comment:
can you put this back before the `sign_checkpoint` call? the intent was to get the duration it takes to sign